src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -24,11 +23,31 @@ namespace MongoDB.Driver.Core.Misc
     /// </summary>
     public class SemanticVersion : IEquatable<SemanticVersion>, IComparable<SemanticVersion>
     {
+        #region static
+        private static void LookForPreReleaseNumericSuffix(string preRelease, out string preReleasePrefix, out int? preReleaseNumericSuffix)
+        {
+            var pattern = @"^(?<prefix>.*[^\d])(?<numericSuffix>\d+)$";
```

Comment:
Consider whether this almost identical regex is slightly more correct:

    var pattern = @"^(?<prefix>[^\d]+)(?<numericSuffix>\d+)$";

If you agree change ServerVersion.cs also.

Reply:
The difference is whether something like `build28jul2020` should match the regex or not.